docs/taskruns.md

```diff
@@ -323,7 +323,7 @@ The following tables shows how to read the overall status of a `TaskRun`:
 Unknown|Started|No|The TaskRun has just been picked up by the controller.
 Unknown|Pending|No|The TaskRun is waiting on a Pod in status Pending.
 Unknown|Running|No|The TaskRun has been validate and started to perform its work.
-Unknown|TaskRunCancelled|No|The user requested the TaskRun to be cancelled. Cancellation has not be done yet.
+Unknown|TaskRunCancelled|Yes|The user requested the TaskRun to be cancelled. Cancellation has not be done yet.
```

Comment:
Uhm, this would be surprising. Did you find a case where `status` was `Unknown` and `completionTime` not set? 

I'm asking because the controller always sets the `completionTime` and the `status` at the same time - or at least that's what's supposed to do. 

Specifically for cancellation, when the user requests cancellation, it may be that it is successful right away - in which case the `status` is set to `False` and `completionTime` is set - see L331 below.
It maybe that cancellation is not possible right away, in which case the reason will still be set to `TaskRunCancelled`, but `status` will stay to `Unknown` and `completionTime` will not be set.
I don't have a reproducer handy for this condition, the most common case will be that in L331 - but this is still a possibility.

Reply:
you are right, this was a mistake, the COMPLETIONTIME is never set as long as the SUCCEEDED column is `Unknown`, see:

```
$ k get taskruns --all-namespaces -w
NAMESPACE        NAME                            SUCCEEDED   REASON   STARTTIME   COMPLETIONTIME
test-build-201   buildrun-test-build-201-pvzh8
test-build-201   buildrun-test-build-201-pvzh8
test-build-201   buildrun-test-build-201-pvzh8   Unknown     Pending   0s
test-build-201   buildrun-test-build-201-pvzh8   Unknown     Pending   0s
test-build-201   buildrun-test-build-201-pvzh8   Unknown     Pending   1s
test-build-201   buildrun-test-build-201-pvzh8   Unknown     Pending   2s
test-build-201   buildrun-test-build-201-pvzh8   Unknown     Pending   3s
test-build-201   buildrun-test-build-201-pvzh8   False       TaskRunCancelled   3s          0s
```

@afrittoli pls close the pr.